### DISCUSSION TITLE
How can I check if PowerShell runs in a Windows Terminal that supports ANSI/VT100 escape codes?

### DISCUSSION THREAD

**Discussion by danielniccoli** [microsoft/terminal#13971] [General]
How can I check if PowerShell runs in a Windows Terminal that supports ANSI/VT100 escape codes?

**Comment by zadjii-msft** [comment]
At the risk of oversimplifying - If you're running on Windows 10, then the console supports escape sequences.

**Reply by DHowett** [reply]
(This will be true regardless of the presence of Windows Terminal!)

**Reply by danielniccoli** [reply]
Huh, you're right. I will tell you what confused me. I noticed that a terminal application I produces bat output when opened in the legacy Windows Console Host.

![grafik](https://user-images.githubusercontent.com/2971735/189717680-aa8b9681-cb82-49a3-847d-1af4a9d8f047.png)

The same command in Windows Terminal produces correct output.

![grafik](https://user-images.githubusercontent.com/2971735/189717810-8e84003b-419b-405b-bfae-86df3ca6018d.png)

Now that you told me, Windows 10 supports ANSI escape codes regardless, I tried to verify this and you're right.

In a PowerShell windows I ran the following line to store an escaped sequence in a bat file: ``"echo \e[5;36mHello World!`e[0m" > test.bat``

In Windows Console Host:

![grafik](https://user-images.githubusercontent.com/2971735/189718462-1644a372-1ccc-438e-981b-1e4c70abb97b.png)

In Windows Terminal:

![grafik](https://user-images.githubusercontent.com/2971735/189718547-b982de57-36d4-415a-ae24-073317c3ddfe.png)

What could be the reason for this behaviour?

**Reply by zadjii-msft** [reply]
[`SetConsoleMode`](https://docs.microsoft.com/en-us/windows/console/setconsolemode), with `ENABLE_VIRTUAL_TERMINAL_PROCESSING`. The Terminal enables it by default for commandline applications. The vintage console does not. `pwsh` I believe also leaves it on by default for client applications, even in conhost.

**Reply by danielniccoli** [reply]
Thank you for the explanation!
